compiler/rustc_codegen_ssa/src/back/symbol_export.rs

```diff
@@ -23,10 +23,8 @@ pub fn threshold(tcx: TyCtxt<'_>) -> SymbolExportLevel {
 
 fn crate_export_threshold(crate_type: CrateType) -> SymbolExportLevel {
     match crate_type {
-        CrateType::Executable | CrateType::Staticlib | CrateType::ProcMacro | CrateType::Cdylib => {
-            SymbolExportLevel::C
-        }
-        CrateType::Rlib | CrateType::Dylib => SymbolExportLevel::Rust,
+        CrateType::Executable | CrateType::Staticlib | CrateType::ProcMacro => SymbolExportLevel::C,
+        CrateType::Rlib | CrateType::Dylib | CrateType::Cdylib => SymbolExportLevel::Rust,
```

Comment:
This can cause conflicts when you link two cdylibs into the same process.

(accidentally commented on commit instead of PR before)

Reply:
Ok so it looks like rust crates can't depend on cdylibs so this is now out of scope. (Have rolled back the change)